torch/_inductor/lowering.py

```diff
@@ -1863,12 +1863,24 @@ def scatter(x, dim: int, index, src, **kwargs):
 
 @register_lowering(aten.scatter_, type_promotion_kind=None)
 def scatter_(self, dim: int, index, src, *, reduce: str = None):
+
+    # TODO: Need to support more reduction type
+    # For reduction of "sum", tl.atomic_add doesn't support bool or int64
+    if reduce not in {None, "add"} or (
+        reduce == "add" and self.get_dtype() in {torch.bool, torch.int64}
+    ):
+        self.realize()
+        return fallback_scatter_(
+            self, dim, index, src, reduce=reduce
+        )
+
     if reduce == "add":
         reduce = "sum"
     elif reduce == "multiply":
         reduce = "prod"
     else:
         assert reduce is None
+
     return scatter_reduce_(self, dim, index, src, reduce)
```

Comment:
This still doesn't handle the non-fallback case with `src` as scalar, right?

Reply:
Ok, our lowering of `scatter_reduce_` does allow `src` to be a number. It's only the fallback causing a problem. LGTM.